pycode/memilio-epidata/memilio/epidata/getVaccinationData.py

```diff
@@ -855,36 +908,72 @@ def get_vaccination_data(read_data=dd.defaultDict['read_data'],
                        directory, filename, file_format)
+
+    # store data for all federal states: group information on date, state and age level 
+    # (i.e., aggregate information of all counties per federal state)
+    filename = 'all_states_ageinf_vacc'
+    filename = gd.append_filename(filename, impute_dates, moving_average)
+    df_data_ageinf_state_cs = df_data_ageinf_county_cs.groupby(
+        [dd.EngEng['date'], dd.EngEng['idState'], dd.EngEng['ageRKI']]).agg({column: sum for column in column_names_new}).reset_index() 
+    gd.write_dataframe(df_data_ageinf_state_cs,
+                       directory, filename, file_format)   
+                          
     end_time = time.perf_counter()
     print("Time needed for age extrapolation: " +
           str(int(end_time - start_time)) + " sec")
     print("Error check time " + str(err_chk_time))
 
     # make plot of relative numbers of original and extrapolated age resolution
     if make_plot:
+        # merge Eisenach...
+        population_new_ages = geoger.merge_df_counties_all(
+            population_new_ages, sorting=[dd.EngEng["idCounty"]],
+            columns=dd.EngEng["idCounty"])
+
+        # have a look extrapolated vaccination ratios (TODO: create plotting for production)
+        # aggregate total number of vaccinations per county and age group
+        latest_date = datetime.strptime(df_data_ageinf_county_cs[dd.EngEng["date"]][len(
+            df_data_ageinf_county_cs.index)-1], "%Y-%m-%d")
+        vacc_sums = df_data_ageinf_county_cs.loc[(
+            df_data_ageinf_county_cs.Date == latest_date.strftime("%Y-%m-%d")), [ dd.EngEng['idCounty'], column_names_new[1]]]
+        df_fullsum_county = compute_vaccination_ratios(unique_age_groups_new, vacc_sums, column_names_new[1], dd.EngEng['idCounty'], population_new_ages, merge_2022=True)
+
+        # have a look extrapolated vaccination ratios (TODO: create plotting for production)
+        # aggregate total number of vaccinations per county and age group
+        latest_date = datetime.strptime(df_data_ageinf_state_cs[dd.EngEng["date"]][len(
+            df_data_ageinf_state_cs.index)-1], "%Y-%m-%d")
+        vacc_sums = df_data_ageinf_state_cs.loc[(
+            df_data_ageinf_state_cs.Date == latest_date.strftime("%Y-%m-%d")), [dd.EngEng['idState'], column_names_new[1]]]
+
+        # get population per federal state
+        population_new_ages.insert(loc=1, column=dd.EngEng["idState"], value=population_new_ages[dd.EngEng["idCounty"]])
+        county_to_state = geoger.get_countyid_to_stateid_map(merge_eisenach=True)
+        for countyid in population_new_ages[dd.EngEng["idState"]].unique():
+            population_new_ages.loc[population_new_ages[dd.EngEng["idState"]]==countyid, dd.EngEng["idState"]] = county_to_state[countyid]
+        population_new_ages_states = population_new_ages.groupby(dd.EngEng['idState']).agg({age_group : sum for age_group in unique_age_groups_new}).reset_index()
```

Comment:
These are all very long lines and can be splitted.

Reply:
done